### ISSUE TITLE
Randomly Logged Out of Desktop App

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#3671]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Was idling. Got logged out.

## Expected Result:
Stay logged in.

## Actual Result:
Logged out.

## Workaround:
Sign back in

## Platform:
 Desktop App

**Version Number:** Version 1.0.70-8 (1.0.70-8)
**Logs:**

<img width="1065" alt="Screen Shot 2021-06-17 at 3 26 09 PM" src="https://user-images.githubusercontent.com/32969087/122491962-a838f280-cf80-11eb-95f7-d5eb0109373e.png">

**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @laurenreidexpensify (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by marcaaron** [comment]
Looks possibly related to the app trying to update itself.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NicMendonca (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by laurenreidexpensify** [comment]
@NicMendonca melvin loves giving me assignments 5min before OOO 😅 reassigning thanks

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mountiny (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by mountiny** [comment]
No update today!

**Comment by mountiny** [comment]
@marcaaron Can this be replicated on dev version? I am using the Desktop app on my Mac and it has never signed off. 

Sorry if it is clear and I am missing something here but do you have any hints, how to reproduce it locally? Thank you very much! :)

**Comment by marcaaron** [comment]
Nope. I can't reproduce it and it has only happened once.

**Comment by rdjuric** [comment]
@mountiny This looks related to https://github.com/electron/electron/issues/7476

Shouldn't happen on prod if our packager is signing the app, but if we don't want this to happen in dev we should sign `node_modules/.../Electron.app` in our scripts

**Comment by marcaaron** [comment]
To clarify, this was experienced in the release build of the app and not while on dev.

**Comment by mountiny** [comment]
Since the issue seems non-reproducible at the moment and happened only once as far as we know, I will degrade its priority label to `Monthly` and we will see if this error occurs again in next month, potentially closing this issue later if no new occurrences happen.

**Comment by mallenexpensify** [comment]
Just happened to me, was in the middle of typing a message, logs below.  Desktop Version 1.0.73-3 (1.0.73-3) (most recent production version). 

```
size 
Object
app-74d3ea1….bundle.js:143 Redirecting to Sign In because we failed to reauthenticate 
Object
app-74d3ea1….bundle.js:143 [Pusher] Attempted to unsubscribe or unbind from a channel,
        but Pusher-JS has no knowledge of it 0 
Object
app-74d3ea1….bundle.js:143 [Pusher] Unable to reconnect since Pusher instance does not yet exist.
w_5e2c6fdc6944ad66e1…e661.png.1024.jpg:1 Failed to load resource: the server responded with a status of 500 ()
app-74d3ea1….bundle.js:143 Unable to fetch image to calculate size 
Object
w_0faa3ead2911202b3f…e78a.png.1024.jpg:1 Failed to load resource: the server responded with a status of 500 ()
app-74d3ea1….bundle.js:143 Unable to fetch image to calculate size 
Object
DevTools failed to load SourceMap: Could not parse content for app://-/app-74d3ea1….bundle.js.map: Unexpected end of JSON input
DevTools failed to load SourceMap: Could not parse content for app://-/pdf.worker.entry-b22bab8….bundle.worker.js.map: Unexpected end of JSON input
```

**Comment by mountiny** [comment]
This might be actually related to Servers being down now.

**Comment by parasharrajat** [comment]
Yeah. I faced it too. Server is down.

**Comment by mallenexpensify** [comment]
If the servers are down, is there a way that we can have e.cash default to offline mode so that it doesn't log users out?

**Comment by mallenexpensify** [comment]
@marcaaron do you think there should be separate issue for "When site is down, go into offline mode"? If so, I can create one

**Comment by parasharrajat** [comment]
Kicking user out without any notification is very bad UX for me.

**Comment by marcaaron** [comment]
>@marcaaron do you think there should be separate issue for "When site is down, go into offline mode"? If so, I can create one

@mallenexpensify yes, I think so. I'm not sure if what you and I experienced are the same. The error message you're seeing is different from the one I originally shared here. Also, IIRC the site was working fine when I was logged out last time.

**Comment by parasharrajat** [comment]
yeah, it seems that this issue is different from the one we are facing today.

**Comment by mallenexpensify** [comment]
Created a separate internal issue to solely focus on 'site down > user logged out' https://github.com/Expensify/Expensify/issues/168965

**Comment by isagoico** [comment]
@puneetlath Was able to reproduce this https://expensify.slack.com/archives/C01GTK53T8Q/p1625069032270200 
___ 
> just got signed out of Expensify.cash on mac desktop unexpectedly

**Comment by jboniface** [comment]
I'm going to put this in `planning` while we wait to see if matt's internal issue resolves it

**Comment by marcaaron** [comment]
This happened to me again similar logs.

```
10:40:37.844 › Checking for update
10:40:38.303 › Update for version 1.0.79-1 is not available (latest version: 1.0.79-1, downgrade is disallowed).
11:40:37.722 › Checking for update
11:40:38.337 › Update for version 1.0.79-1 is not available (latest version: 1.0.79-1, downgrade is disallowed).
3app-ee639f401a1c0000b5d7.bundle.js:143 [LOCAL_NOTIFICATION] Creating notification
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Unsubscribing from channel true Object
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Attempting to subscribe to channel true Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Objectcommand: "Report_UpdateLastRead"doNotRetry: falseisQueuePaused: true__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Objectcommand: "Report_UpdateLastRead"doNotRetry: falseisQueuePaused: true__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Objectcommand: "Report_UpdateLastRead"doNotRetry: falseisQueuePaused: true__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Objectcommand: "Report_UpdateLastRead"doNotRetry: falseisQueuePaused: true__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Objectcommand: "Report_UpdateLastRead"doNotRetry: falseisQueuePaused: true__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 Redirecting to Sign In because we failed to reauthenticate Object
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Attempted to unsubscribe or unbind from a channel,
        but Pusher-JS has no knowledge of it 0 Object
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Attempting to disconnect from Pusher before initialisation has occurred, ignoring.
app-ee639f401a1c0000b5d7.bundle.js:143 A request was made without an authToken Object
app-ee639f401a1c0000b5d7.bundle.js:143 Skipping request and re-queueing:  Object
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Attempting to disconnect from Pusher before initialisation has occurred, ignoring.
app-ee639f401a1c0000b5d7.bundle.js:143 Redirecting to Sign In because we failed to reauthenticate Objectcommand: "Push_Authenticate"error: "passwordForm.error.unableToResetPassword"__proto__: Object
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Unable to reconnect since Pusher instance does not yet exist.
app-ee639f401a1c0000b5d7.bundle.js:143 [Pusher] Attempting to disconnect from Pusher before initialisation has occurred, ignoring.
app-ee639f401a1c0000b5d7.bundle.js:143 A request was made without an authToken Objectcommand: "Report_GetHistory"parameters: {}__proto__: Object
DevTools failed to load SourceMap: Could not parse content for app://-/app-ee639f401a1c0000b5d7.bundle.js.map: Unexpected end of JSON input
DevTools failed to load SourceMap: Could not parse content for app://-/pdf.worker.entry-b22bab8941f6233eb7c1.bundle.worker.js.map: Unexpected end of JSON input
```

**Comment by MelvinBot** [comment]
@marcaaron, this ***Monthly*** task hasn't been acted upon in 6 weeks; closing.

If you disagree, feel encouraged to reopen it -- but pick your least important issue to close instead.

**Comment by marcaaron** [comment]
Still happening

https://expensify.slack.com/archives/C01GTK53T8Q/p1633027687314200

**Comment by marcaaron** [comment]
Maybe we can keep this closed.
